examples/htu_beamline/htu_lattice.py

```diff
@@ -0,0 +1,610 @@
+    ]
+
+    B0 = [
+        SrcToPMQ1,
+        *PMQTrip,
+        PMQTripToTCPhos,
+        TCPhosphor,
+        TCPhosToChicane,
+        S1,
+        *Chicane,
+        S2,
+        DriftToDCPhos,
+        DCPhosphor,
+        DriftToEMQTrip,
+        *EMQTriplet,
+        S3,
+        DriftToPhos1,
+        Phosphor1,
+        DriftToSpec,
+        MagSpec,
+        S4,
+        *Aline,
+    ]
+    Undulator = (
+        UndulatorSegment1 + UndulatorSegment2 + UndulatorSegment3 + UndulatorSegment4
+    )
+    full_beamline = B0 + Undulator
+
+    # Only return the elements between from_element and to_element
+    if from_element is not None:
+        element_indices = [
+            i for i, elem in enumerate(full_beamline) if elem.name == from_element
+        ]
+        if len(element_indices) == 0:
+            raise ValueError(f"Element {from_element} not found in beamline.")
+        elif len(element_indices) > 1:
+            raise ValueError(
+                f"Multiple elements with name {from_element} found in beamline."
+            )
+        # Get the first index of the element
+        from_index = element_indices[0]
+    else:
+        from_index = 0
+    if to_element is not None:
+        element_indices = [
+            i for i, elem in enumerate(full_beamline) if elem.name == to_element
+        ]
+        if len(element_indices) == 0:
+            raise ValueError(f"Element {to_element} not found in beamline.")
+        elif len(element_indices) > 1:
+            raise ValueError(
+                f"Multiple elements with name {to_element} found in beamline."
+            )
+        # Get the first index of the element
+        to_index = element_indices[0]
+    else:
+        to_index = len(full_beamline) - 1
+    beamline = full_beamline[from_index : to_index + 1]
+
+    return beamline
+
+
+# Define a screen element
+def screen(
+    name, code, experiment_configuration_file=None, experiment_analysis_settings=True
+):
+    """
+    Define a screen element.
+    """
+    # If the experimental configuration file is provided, extract the crosshair coordinates, image size and spatial calibration
+    if experiment_configuration_file is not None:
+        with open(experiment_configuration_file, "r") as f:
+            text = f.read()
+        crosshair_x, crosshair_y = extract_crosshair_coordinates(name, text)
+        spatial_calibration = extract_spatial_calibration(name, text)
+        image_size_x, image_size_y = extract_image_size(name, text)
+        # If the experiment analysis settings are given, crop the image
+        if experiment_analysis_settings is not None:
+            # When cropping, we need to measure the count the position
+            # of the crosshair with respect to the edge of the *cropped* image
+            crosshair_x -= experiment_analysis_settings[name]["Left ROI"]
+            crosshair_y -= experiment_analysis_settings[name]["Top ROI"]
+            # Use the cropped image size
```

Comment:
This file still contains Cheetah calls that need to be cleaned :)

Reply:
These calls are not used, and `run_impactx.py` runs correctly in our existing ImpactX environment without Cheetah installed.  To remove all references to Cheetah will require more significant restructuring, but if it is worthwhile then I will proceed.